SCons/Environment.xml

```diff
@@ -486,41 +486,153 @@ Multiple targets can be passed in to a single call to
 </arguments>
 <summary>
 <para>
-Appends the specified keyword arguments
-to the end of construction variables in the environment.
-If the Environment does not have
-the specified construction variable,
-it is simply added to the environment.
-If the values of the construction variable
-and the keyword argument are the same type,
-then the two values will be simply added together.
-Otherwise, the construction variable
-and the value of the keyword argument
-are both coerced to lists,
-and the lists are added together.
-(See also the &Prepend; method).
+Intelligently append values to &consvars; in the &consenv;
+named by <varname>env</varname>.
+The &consvars; and values to add to them are passed as 
+<parameter>key=val</parameter> pairs (Python keyword arguments).
+&f-env-Append; is designed to allow adding values
+without normally having to know the data type of an existing &consvar;.
+Regular Python syntax can also be used to manipulate the &consvar;,
+but for that you must know the type of the &consvar;:
+for example, different Python syntax is needed to combine
+a list of values with a single string value, or vice versa.
+Some pre-defined &consvars; do have type expectations
+based on how &SCons; will use them,
+for example &cv-link-CPPDEFINES; is normally a string
+or list of strings, while &cv-link-LIBEMITTER;
```

Comment:
or a dictionary, or a list of tuples.. right?

Reply:
Sure it can be... I was trying to get away with "normally".

Reply:
Maybe can be a string, a list of strings, a list of tuples, a dictionary, but typically is a string or a list of strings?

Reply:
maybe: "...normally a string or list of strings (a list of tuples or a dictionary are other choices)"